Source/Runtime/EmscriptenIntrinsics.cpp

```diff
@@ -224,6 +224,17 @@ namespace Runtime
 	{
 		return ungetc(character,vmFile(file));
 	}
+	DEFINE_INTRINSIC_FUNCTION4(env,_fread,_fread,I32,I32,pointer,I32,size,I32,count,I32,file)
+	{
+		if(pointer + uint64(size) * (count + 1) > instanceAddressSpaceMaxBytes)
```

Comment:
Hmmmm and really I could overflow it by passing in size 2^32 and count 2^32.  The whole statement needs to be changed.  As does _fwrite which has the same issue.